### PR TITLE
[MRG] Replace `np.in1d()` with `np.isin()`

### PR DESCRIPTION
**Pull request by ntolley**
Problem raised in #794 that numpy version 2.0.0 has deprecated `np.in1d()`

### CODE REVIEW

**Comment by dylansdaniels** [comment]
@ntolley i can squash and merge per our discussion yesterday if we're ready to go. seems like a straightforward substitution

**Comment by rythorpe** [comment]
Looks like there might be some flake8 errors, though it looks like don't have anything to do with this PR. Hmmm...

**Comment by jasmainak** [comment]
should you also bump up the minimum numpy version?

**Comment by ntolley** [comment]
@rythorpe fixed the flake8 errors!

Also @jasmainak no need, `np.isin()` is already a part of numpy v1.14
https://numpy.org/doc/1.14/reference/generated/numpy.isin.html#numpy.isin

File: doc/whats_new.rst [hunk 1]
```diff
@@ -68,6 +68,9 @@ Changelog
   and :func:`~plot_spikes_raster` now plotted from 0 to tstop. Inputs tmin and tmax are deprecated,
   by `Katharina Duecker`_ in :gh:`769`
 
+- Use `np.isin()` in place of `np.in1d()` to address numpy deprecation,
+  by `Nick Tolley`_ in :gh:`799
```
Comment (rythorpe) [inline doc/whats_new.rst:72]:
```suggestion
  by `Nick Tolley`_ in :gh:`799.
```

Comment (dylansdaniels) [reply doc/whats_new.rst:72]: should be `799`. adding the "`" after 799 as well, no? seems the use of the "." at the end is not done consistently in the doc

**Comment by gtdang** [comment]
@ntolley Seems like all tests passed except the linux. Shall we merge this so @kmilo9999 can continue testing the CI updates?

File: doc/whats_new.rst (hunk 1 above)
Comment (jasmainak) [reply doc/whats_new.rst:72]: shouldn't this go in the Bug section? It's not really a new feature

**Comment by jasmainak** [comment]
Just one tiny comment, otherwise good to merge.

**Comment by ntolley** [comment]
> Just one tiny comment, otherwise good to merge.

sounds good! everything should be addressed, whoever seeing the checks (besides ubuntu) go green feel free to merge

**Comment by jasmainak** [comment]
Thanks @ntolley !! 🥳

**Pull request merged**